### ISSUE TITLE
UnnecessaryWarningSuppression should report which suppression is unnecessary

### ISSUE THREAD

**Issue by aaime** [pmd/pmd#5786]
**Is your feature request related to a problem? Please describe.**

UnnecessaryWarningSuppression just reports that an unnecessary suppression occurs, and on which line. But a SuppressWarnings may contain multiple suppression, it's not clear which one(s) could be removed.

**Describe the solution you'd like**
In the error message, report which suppressions are unnecessary

**Describe alternatives you've considered**
Bisect and rerun maven, but it can be annoying if the module in question is large.

(btw, UnnecessaryWarningSuppression is great, findings lots of unnecessary annotations)

**Comment by oowekyala** [comment]
I think we already try to do that. If the annotation is reported then all its suppressions are unnecessary, otherwise the message is like `unnecessary suppression "PMD.ARule" in @SuppressWarnings annotation`. Can you share a specific example where the behavior is problematic?

**Comment by aaime** [comment]
I've tried again and indeed the message is there... doh been staring at the XML reports generated by the maven plugin all afternoon and did not notice. I'm used to identify the issue by looking at the beginning of the XML element and failed to see the message at the end indeed reports the issue:

```xml
<file name="/home/aaime/devel/git-gt/modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCDataStoreOnlineTest.java">
<violation beginline="94" endline="94" begincolumn="24" endcolumn="51" rule="UnnecessaryWarningSuppression" ruleset="Best Practices" package="org.geotools.jdbc" class="JDBCDataStoreOnlineTest" method="testCreateSchema" externalInfoUrl="https://docs.pmd-code.org/pmd-doc-7.14.0/pmd_rules_java_bestpractices.html#unnecessarywarningsuppression" priority="3">
Unnecessary suppression "PMD.EmptyControlStatement" in @SuppressWarnings annotation
</violation>
</file>
```

Never mind, and sorry for the noise!
